"Create drilldown" flyout - design cleanup

andreadelrio opened the pull request:
## Summary

Some improvements to the flyout. I'll fix the aligment issue (banner's content with flyout's header) directly on Eui. 

![image](https://user-images.githubusercontent.com/4016496/76795595-d67cc080-6786-11ea-861c-657e2db1f7ef.png)


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server)
- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

streamich commented:
Curious, why are we removing the rounded corner grey background around the dashboard configuration? It looked so nice with the background.

@andreadelrio @mdefazio

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/)
* Commit: 5dbf4d32f9049dfc315238c74c7c8b3633d34d31

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/testReport)

<details><summary>Kibana Pipeline / kibana-intake-agent / Jest Tests.src/plugins/embeddable/public/lib/panel.HelloWorldContainer in edit mode hides disabledActions</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/testReport//junit/Jest%20Tests/src_plugins_embeddable_public_lib_panel/Kibana_Pipeline___kibana_intake_agent___HelloWorldContainer_in_edit_mode_hides_disabledActions)


#### Standard Out
```
Failed Tests Reporter:
  - Test has not failed recently on tracked branches


```

#### Stack Trace
```
Error: expect(received).toBe(expected) // Object.is equality

Expected: 2
Received: 1
    at Object.<anonymous>.test (/var/lib/jenkins/workspace/elastic+kibana+pipeline-pull-request/kibana/src/plugins/embeddable/public/lib/panel/embeddable_panel.test.tsx:247:44)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```
</details>

---

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/","number":"33579","commit":"5dbf4d32f9049dfc315238c74c7c8b3633d34d31"}]}
PIPELINE-->

andreadelrio commented:
> Curious, why are we removing the rounded corner grey background around the dashboard configuration? It looked so nice with the background.
> 
> @andreadelrio @mdefazio

@streamich We're trying to be as consistent as possible with what we have for Alerting.

![alerting-s](https://user-images.githubusercontent.com/4016496/76802008-f797de00-6793-11ea-9a60-c4862285ff2b.gif)

mdefazio commented:
I think this is good, though I think we can remove items which really only help when we have the selection screen at the beginning 

Current:
![image](https://user-images.githubusercontent.com/3756330/76814003-ff757380-67cf-11ea-9f04-502483323905.png)

Update:
![image](https://user-images.githubusercontent.com/3756330/76813935-d7861000-67cf-11ea-8181-e23f66fb7f8e.png)

I think we can remove 'Drilldown action' so there aren't so many labels.  I also think we can increase the space between the dropdown and the switches.

Perhaps we also make the dashboard dropdown full width?

Ignore the copy changes for now, we can tackle these in their own PR as I think there are multiple places we need to update.

andreadelrio commented:
> I think this is good, though I think we can remove items which really only help when we have the selection screen at the beginning
> 
> Current:
> ![image](https://user-images.githubusercontent.com/3756330/76814003-ff757380-67cf-11ea-9f04-502483323905.png)
> 
> Update:
> ![image](https://user-images.githubusercontent.com/3756330/76813935-d7861000-67cf-11ea-8181-e23f66fb7f8e.png)
> 
> I think we can remove 'Drilldown action' so there aren't so many labels. I also think we can increase the space between the dropdown and the switches.

> Ignore the copy changes for now, we can tackle these in their own PR as I think there are multiple places we need to update.

@mdefazio Thanks for catching those. I updated the labels and removed the unneeded "Drilldown action" one, so it now it looks like the screenshot you have under "Update". I was unable to add more space between the dashboard picker and the switches though. I'll leave that one to @Dosant or @streamich 


> 
> Perhaps we also make the dashboard dropdown full width?
> 

Regarding this, there's a [PR](https://github.com/elastic/kibana/pull/60087) in progress for it. This is the screenshot that Vadim share of it on the Slack group 

![image (7)](https://user-images.githubusercontent.com/4016496/76819805-8a517080-67c6-11ea-896c-1ced099dca3f.png)

kibanamachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33688/)
* Commit: 374846f1dba9b8cfefca1993eefd796def42ac79

### History
* :broken_heart: [Build #33579](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/) failed 5dbf4d32f9049dfc315238c74c7c8b3633d34d31

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/","number":"33579","commit":"5dbf4d32f9049dfc315238c74c7c8b3633d34d31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33688/","number":"33688","commit":"374846f1dba9b8cfefca1993eefd796def42ac79"}]}
PIPELINE-->

Dosant commented:
I pushed a type check fix. Also didn't remove "action" label complete, but just left it for case when there is more then 1 action factory (for future). 
Hope that's ok

kibanamachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33731/)
* Commit: d56131e8cc730f07d3f72095858e43bf412ebf40
* Flaky suites:
  * xpack-siemCypress

---

### [Test Failures](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33731/testReport)

<details><summary>Kibana Pipeline / kibana-xpack-agent / adds a field to the timeline when the user drags and drops a field.Fields Browser Editing the timeline adds a field to the timeline when the user drags and drops a field</summary>

[Link to Jenkins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33731/testReport//junit/(root)/adds%20a%20field%20to%20the%20timeline%20when%20the%20user%20drags%20and%20drops%20a%20field/Kibana_Pipeline___kibana_xpack_agent___Fields_Browser_Editing_the_timeline_adds_a_field_to_the_timeline_when_the_user_drags_and_drops_a_field)


#### Standard Out
```
Failed Tests Reporter:
  - Test has failed 2 times on tracked branches: https://github.com/elastic/kibana/issues/60023


```

#### Stack Trace
```
CypressError: Timed out retrying: Expected to find element: '[data-test-subj="header-text-host.geo.country_name"]', but never found it.
    at cypressErr (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:149934:9)
    at throwErr (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:149867:11)
    at Object.throwErrByPath (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:149915:3)
    at retry (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:144197:19)
    at onFailFn (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:132448:16)
    at tryCatcher (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:167415:23)
    at Promise._settlePromiseFromHandler (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:165351:31)
    at Promise._settlePromise (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:165408:18)
    at Promise._settlePromise0 (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:165453:10)
    at Promise._settlePromises (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:165528:18)
    at Async../node_modules/bluebird/js/release/async.js.Async._drainQueue (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:162140:16)
    at Async../node_modules/bluebird/js/release/async.js.Async._drainQueues (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:162150:10)
    at Async.drainQueues (http://elastic:changeme@localhost:61121/__cypress/runner/cypress_runner.js:162024:14)
```
</details>

---

### History
* :broken_heart: [Build #33688](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33688/) failed 374846f1dba9b8cfefca1993eefd796def42ac79
* :broken_heart: [Build #33579](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/) failed 5dbf4d32f9049dfc315238c74c7c8b3633d34d31

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33579/","number":"33579","commit":"5dbf4d32f9049dfc315238c74c7c8b3633d34d31"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33688/","number":"33688","commit":"374846f1dba9b8cfefca1993eefd796def42ac79"},{"status":"UNSTABLE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/33731/","number":"33731","commit":"d56131e8cc730f07d3f72095858e43bf412ebf40"}]}
PIPELINE-->
